src/cli/index.js

```diff
@@ -107,7 +107,7 @@ export function main({
     commandName = 'help';
   }
 
-  if (args.indexOf('--help') >= 0 || args.indexOf('-h') >= 0) {
+  if (Object.prototype.hasOwnProperty.call(commands, commandName) && (args[0] === '--help' || args[0] === '-h')) {
```

Comment:
Not strongly opinionated on this tho, so if you prefer it as it is I'm fine with it

Reply:
I thought about that but that approach prohibits running `yarn run --help`, right?

Reply:
I wouldn't worry too much about this case, but yeah, it won't work.

Reply:
I'd rather keep it in case we introduce more flags for `run` and to keep `--help` behavior consistent. Are you okay if we keep it the way it is in the patch?

Reply:
Sure 👍